### PR TITLE
CLE-NST-ASCT - Add new Supported Container Types

### PR DESCRIPTION
**Pull request by VLADISLAV008**

### CODE REVIEW

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 1]
```diff
@@ -3,18 +3,23 @@ package group.scala.karazin.circe.literal.extras
...
 import scala.quoted._
-
 import scala.deriving.Mirror
+import scala.util.{Failure, Success, Try as UTry}
 
 object macros:
 
   object encode:
 
+    final case class EncodeException(private val message: String = "") extends Exception(message)
```
Comment (IgorWolkov) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: Usually case classes do not have private values.
Also according to our logic the message is required.
So let's make it just `EncodeException(message: String)`.

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 2]
```diff
@@ -3,18 +3,23 @@ package group.scala.karazin.circe.literal.extras
...
+import cats.implicits._
+import cats.data.{Chain, NonEmptyChain, NonEmptyList, NonEmptyMap, NonEmptySet, NonEmptyVector, OneAnd, Validated}
 import io.circe.parser
 import io.circe.syntax._
 import io.circe.{Json, JsonObject, JsonNumber, Encoder, ACursor, HCursor}
 
 import scala.quoted._
-
 import scala.deriving.Mirror
+import scala.util.{Failure, Success, Try as UTry}
```
Comment (IgorWolkov) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: Haven't found the case of name clash.

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 3]
```diff
@@ -305,46 +377,130 @@ object macros:
...
+            validateJsonSchema[t](key, value.hcursor)
+          }
+
+        case '[NonEmptyVector[t]] =>
+          validateJsonArray(key, cursor, true) { value =>
+            validateJsonSchema[t](key, value.hcursor)
+          }
+    
+        case '[NonEmptySet[t]] =>
+          validateJsonArray(key, cursor, true, true) { value =>
```
Comment (IgorWolkov) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: Always provide field names for hardcoded values:
```scala
validateJsonArray(nonEmpty = true, differentValues = false)
```

**Review by IgorWolkov — COMMENTED** [review]
I've added minor comments, need to verify `validateJsonArray` and `validateJsonObject` carefuly.

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala (hunk 2 above)
Comment (VLADISLAV008) [reply src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: Clash occurs with this import
`import quotes.reflect._`

**Review by IgorWolkov — APPROVED** [review]

**Pull request merged**